File: agent-api/app/services/chat/builtin_assistants/presentation/definition.py
"""Presentation product identity and catalog defaults; no execution dependencies."""

from app.services.chat.builtin_assistants.types import (
    BuiltinAppSpec,
    BuiltinAssistantDefinition,
    BuiltinAssistantIdentity,
)

PRESENTATION_PRESET = "presentation"
PRESENTATION_APP_ID = "builtin:presentation"
PRESENTATION_THREAD_ORIGIN = "presentation"
PRESENTATION_IDENTITY: BuiltinAssistantIdentity = {
    "app_id": PRESENTATION_APP_ID,
    "preset": PRESENTATION_PRESET,
    "name": "演示文稿助手",
    "origin": PRESENTATION_THREAD_ORIGIN,
    "ui_policy_key": "presentation_authoring",
}
PRESENTATION_DEFINITION = BuiltinAssistantDefinition(
    identity=PRESENTATION_IDENTITY,
    catalog=BuiltinAppSpec(
        preset=PRESENTATION_PRESET,
        name=PRESENTATION_IDENTITY["name"],
        description="制作、优化并交付可编辑演示文稿",
        icon="/agent-icons/builtin/presentation-assistant.png",
        category="content_creation",
        category_label="内容创作",
        route="/center/chat/ppt",
        order_num=20,
    ),
)


def is_presentation_preset(value: object) -> bool:
    return str(value or "").strip().lower() == PRESENTATION_PRESET

File: src/views/peopleCenter/builtinAssistants/presentation/definition.ts
import {
  PRESENTATION_ASSISTANT_PRESET,
  type BuiltinAssistant,
} from '../types';

export const PRESENTATION_ASSISTANT_ID = 'builtin:presentation';

export const PRESENTATION_ASSISTANT: BuiltinAssistant = {
  appId: PRESENTATION_ASSISTANT_ID,
  preset: PRESENTATION_ASSISTANT_PRESET,
  name: '演示文稿助手',
  shortBadge: '演示文稿',
  description: '制作、优化并交付可编辑演示文稿',
  icon: '/agent-icons/builtin/presentation-assistant.png',
  legacyIcon: '/agent-icons/presentation-assistant.png',
  category: 'content_creation',
  categoryLabel: '内容创作',
  route: '/center/chat/ppt',
  entryMode: 'standalone',
  recommendOrder: 2,
  mascotVariant: 'presentation',
  uiPolicy: 'presentation_authoring',
  notices: {
    useSkill: '演示文稿助手固定使用 ppt-studio；其他 Skill 请在主对话的新会话中使用',
  },
};
